c2corg_api/views/user_mailinglists.py

```diff
@@ -0,0 +1,48 @@
+from c2corg_api import DBSession
+from c2corg_api.models.user import User
+from c2corg_api.models.mailinglist import Mailinglist
+from c2corg_api.views import cors_policy, restricted_json_view, \
+    restricted_view
+from c2corg_common.attributes import mailinglists
+from cornice.resource import resource
+from colander import MappingSchema, SchemaNode, String, Boolean, Sequence, \
+    OneOf, required
+from sqlalchemy.orm import joinedload, load_only
+
+
+@resource(path='/users/mailinglists', cors_policy=cors_policy)
+class UserMailinglistsRest(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @restricted_view()
+    def get(self):
+        """Get the mailinglists subscriptions of the authenticated user.
+
+        Request:
+            `GET` `/users/mailinglists`
+
+        Example response:
+
+            {'avalanche': False, 'lawinen': True, 'valanghe': False, ...}
+
+        """
+        user_id = self.request.authenticated_userid
+        listnames = DBSession.query(Mailinglist.listname). \
+            options(joinedload(Mailinglist.user).load_only(User.email)). \
+            filter(User.id == user_id). \
+            all()
+
+        return {ml: ml in mailinglists for ml in listnames}
+
+    @restricted_json_view(schema=MailinglistSchema())
+    def post(self):
+        """Update a mailinglist subscription status.
+
+        Request:
+            `POST` `/users/mailinglists`
+            {'avalanche': False, 'lawinen': True, 'valanghe': False}
```

Comment:
> You are planning to do a request every time you toggle the subscription for a list?

Yes, because:
* this way no need to press a "save" button in the UI, which is what modern UI do now (including our preferences page)
* most people only subscribe to a few lists. In Switzerland, most likely only one (or 2 if you live near Haute Savoie). If you leave in Chambery, maybe 3 (mf-74, mf-73, mf-38).
* people don't need to change all there subscriptions at a time, maybe 1 or 2 from times to times, for instance if you plan a trip in a remote place and want to follow the avalanche situation a week before leaving (personally I have never changed my 2-list subscription I think :P)

> Maybe you don't even need an explicit schema. Simply a validator that checks that the list names are one of c2corg_common.attributes.mailinglists could be enough.

OK. I admit I am not really sure what schema are used for. I guess something simple is enough.
Then I will skip the schema and use only a validator if you think a schema is not mandatory here.